### PR TITLE
feat(photodesk): Added photodesk view on monitoring

### PR DESCRIPTION
**Pull request by pavlovicnemanja**
SDANSA-112 - Grid view in desk monitoring view

### CODE REVIEW

File: scripts/apps/monitoring/directives/MonitoringGroup.js [hunk 1]
```diff
@@ -63,6 +63,12 @@ export function MonitoringGroup(cards, api, authoringWorkspace, $timeout, superd
                 });
             });
 
+            scope.$on('view', (event, data) => {
```
Comment (petrjasek) [inline scripts/apps/monitoring/directives/MonitoringGroup.js]: I guess you can watch `monitoring.view` here, these should be in sync right?

Comment (petrjasek) [reply scripts/apps/monitoring/directives/MonitoringGroup.js]: or you could set it via directive scope and pass it there

File: scripts/apps/search/components/PhotoDeskInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            <span className="sd-grid-item__slugline" dangerouslySetInnerHTML={createMarkUp(item.description_text)} />
+            {source}
+        </div>
+    );
+}
+
+PhotoDeskInfo.propTypes = {
+    svc: PropTypes.object.isRequired,
+    item: PropTypes.any,
+    ingestProvider: PropTypes.any
```
Comment (tomaskikutis) [inline scripts/apps/search/components/PhotoDeskInfo.jsx]: Unused prop

File: scripts/apps/search/components/PhotoDeskPreview.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            onMultiSelect: props.onMultiSelect,
+            svc: props.svc
+        })
+    );
+}
+
+PhotoDeskPreview.propTypes = {
+    svc: PropTypes.object.isRequired,
+    onMultiSelect: PropTypes.func,
+    desk: PropTypes.any,
```
Comment (tomaskikutis) [inline scripts/apps/search/components/PhotoDeskPreview.jsx]: unused props `desk` and `item`

File: scripts/apps/monitoring/controllers/MonitoringController.js [hunk 4]
```diff
@@ -120,7 +121,7 @@ export function MonitoringController($rootScope, $scope, $location, desks, confi
      * @param {Boolean} updateSession if set it will update user session setting.
      * @returns {Number|null} function returns columnsLimit null if viewColumn is false.
      */
-    function switchView(viewColumn, updateSession) {
+    function switchViewColumn(viewColumn, updateSession) {
```
[the remark this answers is not in the document]
Comment (pavlovicnemanja) [reply scripts/apps/monitoring/controllers/MonitoringController.js]: it's not missing, it's over there ` @param {Boolean} viewColumn if set to true then function returns columnsLimit for swimlane as per configuration.`

[the remark this answers is not in the document]
Comment (tomaskikutis) [reply scripts/apps/monitoring/controllers/MonitoringController.js]: Can't see it

c

**Comment by tomaskikutis** [comment]
If you add code after submiting a PR, do so in separate commits so others don't have to review the same code multiple times. You can squash those commits into one when merging :)

File: scripts/apps/search/components/PhotoDeskInfo.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {createMarkUp} from '../helpers';
+
+/**
+ * PhotoDesk Info - renders item metadata
+ */
+export function PhotoDeskInfo(props) {
```
Comment (petrjasek) [inline scripts/apps/search/components/PhotoDeskInfo.jsx]: you can use jsx in other files too, not just here ;)

File: scripts/apps/monitoring/controllers/MonitoringController.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (pavlovicnemanja) [reply scripts/apps/monitoring/controllers/MonitoringController.js]: if  you extend file you can see it. This is old function, jsdoc was already defined here:
image

Comment (tomaskikutis) [reply scripts/apps/monitoring/controllers/MonitoringController.js]: You're right. Didn't see I could extend it.

File: scripts/apps/search/components/PhotoDeskPreview.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        React.createElement(SelectBox, {
+            item: item,
+            classes: 'sd-grid-item__checkbox',
+            onMultiSelect: props.onMultiSelect,
+            svc: props.svc
+        })
+    );
+}
+
+PhotoDeskPreview.propTypes = {
```
Comment (petrjasek) [inline scripts/apps/search/components/PhotoDeskPreview.jsx]: it's missing `item: PropTypes.object,` here

File: scripts/apps/search/components/PhotoDeskInfo.jsx (hunk 5 above)
Comment (pavlovicnemanja) [reply scripts/apps/search/components/PhotoDeskInfo.jsx]: 😑

**Comment by pavlovicnemanja** [comment]
hey @tomaskikutis, we are doing this because our commit's are displayed in release log. And if we firstly create separate commits, and after all reviews are done, we squash them into one, it is from my point of view needless lose of time and task time.

**Comment by petrjasek** [comment]
@pavlovicnemanja what release log you mean?

**Comment by tomaskikutis** [comment]
@pavlovicnemanja squashing the commits when merging is a one-click thing. It definitely takes longer to find that one line you changed when reviewing. You can make as much or little commits as you like locally, I was talking about updating the PRs which are already submitted.

squash-merge

**Comment by pavlovicnemanja** [comment]
@petrjasek , I remembered that release description was generated from merged commit messages... but I checked now, so I guess this is no longer the case.

@tomaskikutis this was not there previously 🙂 , I didn't see it

**Pull request merged**